### PR TITLE
Podman support

### PR DESCRIPTION
**Pull request by ydirson**
Builds on #29

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -20,7 +20,15 @@
...
-RUNNER = os.getenv("XCPNG_OCI_RUNNER", "docker")
+RUNNER = os.getenv("XCPNG_OCI_RUNNER")
+if RUNNER is None:
+    SUPPORTED_RUNNERS = "docker podman"
+    for command in SUPPORTED_RUNNERS.split():
+        if shutil.which(command):
+            RUNNER = command
+            break
+    else:
+        raise Exception(f"cannot found a supported runner: {SUPPORTED_RUNNERS}")
```
Comment (stormi) [inline run.py:31]: "cannot find", or "haven't found"?

File: build.sh [hunk 2]
```diff
@@ -8,7 +8,22 @@ if [ -z "$1" ]; then
...
+else
+    SUPPORTED_RUNNERS="docker podman"
+    for COMMAND in $SUPPORTED_RUNNERS; do
+        if command -v $COMMAND >/dev/null; then
+            RUNNER="$COMMAND"
+            break
+        fi
+    done
+    if [ -z "$RUNNER" ]; then
+        echo >&2 "cannot found a supported runner: $SUPPORTED_RUNNERS"
```
Comment (stormi) [inline build.sh:23]: "cannot find", or "haven't found"?

**Review by stormi — COMMENTED** [review]
LGTM, but what will happen if podman is installed, but also provides the `docker` command, as is the case on Fedora and maybe elsewhere?

**Comment by ydirson** [comment]
> LGTM, but what will happen if podman is installed, but also provides the `docker` command, as is the case on Fedora and maybe elsewhere?

Good point. Would indeed seem to advocate for looking for `podman` before `docker` :smiling_imp: 

It would require identifying that a `docker` binary is indeed `podman`, likely with:
```
$ docker --version 2>/dev/null
podman version 4.7.2
```

But that case already does not work today, this PR just does not address that particular issue (ie. we could handle this in a separate PR).

**Review by stormi — APPROVED** [review]

**Pull request merged**